boilerplate/openshift/golang-osd-operator/.codecov.yml

```diff
@@ -0,0 +1,26 @@
+codecov:
```

Comment:
Do we really want this? I'm not the biggest fan of it due to the lack of accuracy

Reply:
You mean because e.g. removing a line of code makes it look like you're decreasing coverage (e.g. 94/99 < 95/100)? Or because it's not taking generated code into account? Or something else? Is it the go cover tool specifically that you have issues with, or any coverage analysis?

My take is that doing it with inaccuracy is better than not doing it, since the limitations are known and we're not relying on the result at any serious level of precision. Currently we're [setting the bar ridiculously low](https://github.com/openshift/boilerplate/blob/459c3cdf9d1bf451f99a08fbc21833c817d540e1/boilerplate/openshift/golang-osd-operator/.codecov.yml#L8); if the inherent inaccuracies are making the difference between pass and fail, IMO you should be greatly increasing test coverage anyway.

Reply:
are you saying we fail PRs if the lower the test coverage?

Reply:
What I mean is, that I don't see a benefit in having it. Especially given the current coverage

Reply:
> are you saying we fail PRs if the lower the test coverage?

Yes. The cutoff is currently 20%. We hope to be able to raise that as time goes on. But it's understood that there's no expectation to test e.g. generated controller framework code, so it'll never be 100%. (As one example, I feel like [aws-efs-operator](https://codecov.io/gh/openshift/aws-efs-operator) has pretty much all of its non-generated code covered, and it shows just north of 75%.)

> What I mean is, that I don't see a benefit in having it. Especially given the current coverage

Are you saying the current coverage is high or low? (I didn't check.) Or are you saying that the integration tests, which wouldn't run under coverage, obviate the need for unit tests?

In general the benefits include:
- It gives reviewers a hint as to whether you forgot to include unit tests with your PR (if coverage goes down significantly).
- It provides nice visual dashboards that can help you easily find hot spots of covered/uncovered code.

If you want to ignore the reports as a team, that's up to you. If you seriously have <20% unit test coverage currently, you may want to consider remedying that -- integration tests notwithstanding. But the whole point of boilerplate is to avoid snowflaking individual consumers. So I'd really like to keep this in place.

Reply:
A CI should never fail based on code coverage, probably we should fix that in boilerplate. What's the reason to block a PR that lowers the coverage from 20% to 19% while one that lowered it from 30% to 29% is fine? Probably it's a theoretical issue, since we don't have that low coverage in this repository, but I think failing CI based on code coverage is an antipattern. I'm fine with making it visible in the PR conversation, though.

Reply:
The 20% threshold is there to encourage the team to add tests, its not a high bar and we could also change it. I would agree with you @NautiluX if the threshold were > 70% and CI tests were failing all the time. I don't see the hard here. 

@RiRa12621 If you have a specific issue with codecov let us know, sounds like you had some questions around the accuracy but I'm also not sure what you mean.

Reply:
I bet now you're happy about this comment 😜